DRY up help site (#12349)
Expensify/App: docs/README.md

1. Modify a `.html` or `.md` file and save your changes, and the browser should quickly auto-refresh.

# How the project is structured

The [docs](https://github.com/Expensify/App/tree/main/docs) folder will contain the following main folders:

- *_layouts*: Contains HTML templates used by Jekyll to create pages in the static site. For now it has only one file ([default.html](https://github.com/Expensify/App/blob/main/docs/_layouts/default.html)) which will render the common HTML elements and the assets on every page.
- *_includes*: Contains HTML content that can be reused on different pages. [Includes](https://jekyllrb.com/docs/includes/) can accept parameters and be nested, so they are quite powerful.
- *assets*: Contains the three sub-folders css, images, and js. The css folder can contain either .css or .sass files (where SASS files will be processed by Jekyll and it will generate a CSS file in an output folder also generated by Jekyll).
- *hubs*: Contains hubs that display various related articles, optionally grouped by sub-topic within the hub.
- *articles*: Contains sub-folders for each hub and each of them will contain all the articles related to that hub.

> - *_data*: Contains [custom global data](https://jekyllrb.com/docs/datafiles/), available anywhere via liquid with the syntax `site.data.<filename>`. Notably, there is a file called routes.yml in there which defines the structure of the app and metadata for its pages. 

Reviewer: I think it can be handy to link the [routes.yml](https://github.com/Expensify/App/blob/55f2c73d8e3bb0efd0e2a985c05f25e87d28f3cc/docs/_data/routes.yml) here:

```suggestion
- *_data*: Contains [custom global data](https://jekyllrb.com/docs/datafiles/), available anywhere via liquid with the syntax `site.data.<filename>`. Notably, there is a file called [routes.yml](https://github.com/Expensify/App/blob/55f2c73d8e3bb0efd0e2a985c05f25e87d28f3cc/docs/_data/routes.yml) in there which defines the structure of the app and metadata for its pages. 
```

- *_site*: The output folder that contains the generated HTML files. This is the default name by Jekyll but can be changed. Files in here are generated and should not be manually edited.

> - index.html: The main homepage contentx

Reviewer: ```suggestion
- index.html: The main homepage content.
```

More details about the Jekyll project structure can be found [here](https://jekyllrb.com/docs/structure/). Note that all html files in this project support the use of the [liquid template language](https://shopify.github.io/liquid/).

# How to add a new Article

## Add content

> Copy the [template](https://github.com/Expensify/App/blob/main/docs/TEMPLATE.md) file into the correct subdirectory of `/articles`. For example, if the article belongs in the `Send money` hub, then copy the template into `articles/send-money` directory. Next, rename the copy with the name of the article title, i.e. [SmartScan.md](https://github.com/Expensify/App/blob/main/docs/articles/request-money/SmartScan.md) (you can use dashes for spaces in the file name if it's needed) and put the new file inside of the respective hub folder or sub-folder. The title will be rendered automatically according to the filename (the dashes will be removed in the generated site page).

Reviewer: Oh let's use here the `The Free Plan` since `Smart Scan` was removed recently:

```suggestion
Copy the [template](https://github.com/Expensify/App/blob/main/docs/TEMPLATE.md) file into the correct subdirectory of `/articles`. For example, if the article belongs in the `Send money` hub, then copy the template into `articles/send-money` directory. Next, rename the copy with the name of the article title, i.e. [The-Free-Plan.md](https://github.com/Expensify/App/blob/main/docs/articles/send-money/The-Free-Plan.md) (you can use dashes for spaces in the file name if it's needed) and put the new file inside of the respective hub folder or sub-folder. The title will be rendered automatically according to the filename (the dashes will be removed in the generated site page).
```

title: The title of the post, page, or document
description: A short description of the page's content
image: (Optional) URL to an image associated with the post, page, or document (e.g., /assets/page-pic.jpg)
---
```

Just update the content for each variable accordingly or remove it if the information is not available. More details [here](https://github.com/jekyll/jekyll-seo-tag/blob/master/docs/usage.md#usage).

## Add the new page to routes.yml

> Next, add the article to `_data/routes.yml`. Note that hubs contain one or more articles, which may or may not be grouped into sections with other related articles.

Reviewer: 🔥 🤘🏽
